### ISSUE TITLE
Multi-CPU support

### ISSUE THREAD

**Issue by arunoda** [arunoda/meteor-up-legacy#8]

**Comment by copleykj** [comment]
:+1: 

With galaxy getting tabled till 1.1 this would be amazing!

**Comment by stephentcannon** [comment]
+1 crossing similar bridge

**Comment by tonyaom** [comment]
It is a great feature that may make using Meteor for scale projects easier.

Look forward to seeing the Multi-CPU being released.

**Comment by Kikobeats** [comment]
@arunoda I read your post in [meteorhacks](https://meteorhacks.com/run-meteor-on-multiple-cpus.html) and I think that this can be easily to resolve using PM2 (#88) instead of forever because you can set up using the max of CPU's!

**Comment by arunoda** [comment]
I'd love to use it.
But it doesn't support sticky sessions. see:
https://github.com/Unitech/pm2/issues/389

**Comment by Kikobeats** [comment]
hmm, I didn't know how SockJS managed the session, but look that is significantly a problem to be scalable.

do you think that is possible resolve the problem writing a bit of code using node cluster api? people say in the issue that are using a library called [sticky-session](https://github.com/indutny/sticky-session) and later lift the script with pm2.

**Comment by arunoda** [comment]
No, that project does not work for us.
It's based on IP address and that's not a good way to doing sticky sessions.

**Comment by babnik63** [comment]
Yes that project seems not to fit. But this is a critical need, I hope we reach to a solution. :-)

**Comment by strikeout** [comment]
How about using Phusion Passenger in standalone or nginx integration mode?

**Comment by babnik63** [comment]
I think using nginx for load balancing between multiple apps running on different cores could be a good idea, but it needs some standardization and proper management with something like mup.

**Comment by Kikobeats** [comment]
@arunoda what about using a web-worker API? https://www.npmjs.com/package/webworker-threads

**Comment by MaazAli** [comment]
Any update on this. You've recently talked about `meteor-cluster`, will that essentially launch more node processes to support multi-cpu like the node cluster module?

**Comment by arunoda** [comment]
It does not spawn instances. It's different from the node module. This is a
complete solution which is a real proxy and does more work.

**Comment by MaazAli** [comment]
@arunoda maybe this is not the right place to talk about it but will [meteor-cluster](https://github.com/meteorhacks/cluster/) utilize servers with multi-cpus? Load balance between multiple instances running on the same server using one cpu per instance for example.

**Comment by arunoda** [comment]
We might can use it for that. I'll share some patterns soon.

**Comment by katopz** [comment]
Sound interesting, here's just for ref btw https://github.com/andruschka/pm2-meteor/issues/6
